src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/ControlProblemElements.h

```diff
@@ -0,0 +1,745 @@
+     * @param r torso parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @param R torso parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @param t shoulder parameter http://wiki.icub.org/wiki/ICub_coupled_joints
+     * @throw std::runtime_error if the frame is not defined
+     * @note this is iCub specific. In the next version it will be removed
+     */
+    [[deprecated(
+        "This function is iCub specific. In the next implementation will be removed. If "
+        "you want to include regularization element for the mass matrix please call "
+        "SystemDynamicsElement(iDynTree::KinDynComputations&, const VariableHandler&, "
+        "const std::vector<std::pair<std::string, std::string>>&, const "
+        "iDyntree::MatrixDynsize&)")]]
+    SystemDynamicsElement(iDynTree::KinDynComputations& kinDyn,
+                          const VariableHandler& handler,
+                          const std::vector<std::pair<std::string, std::string>>& framesInContact,
+                          const iDynTree::VectorDynSize& gamma,
+                          const iDynTree::VectorDynSize& motorsInertia,
+                          const iDynTree::VectorDynSize& harmonicDriveInertia,
+                          const double& r,
+                          const double& R,
+                          const double& t);
+
+    /**
+     * Constructor. If you call this constructor the motor reflected inertia is enable
+     * @param kinDyn an iDynTree kinDyn computation object
+     * @param handler the variable handler object
+     * @param framesInContact vector containing the frames in contact. Each element of the vector is
+     * a pair containing the name of the frame in the variableHandler and in the model.
+     * @param regularizationMatrix joints regularization mass matrix. It can be used to handle
+     * the motor reflected inertia
+     * @throw std::runtime_error if the frame is not defined or the regularizationMatrix size is
+     * no coherent with the number of joints
+     */
+    SystemDynamicsElement(iDynTree::KinDynComputations& kinDyn,
+                          const VariableHandler& handler,
+                          const std::vector<std::pair<std::string, std::string>>& framesInContact,
+                          const iDynTree::MatrixDynSize& regularizationMatrix);
+
+    /**
+     * Get (and compute) the element matrix
+     * @return the element matrix
+     */
+    virtual const iDynTree::MatrixDynSize& getA() final;
+
+    /**
+     * Get (and compute) the element vector
+     * @return the element vector
+     */
+    virtual const iDynTree::VectorDynSize& getB() final;
+};
+
+/**
+ * CentroidalLinearMomentumElement describes the Centroidal linear momentum
+ */
+class CentroidalLinearMomentumElement : public CostFunctionOrEqualityConstraintElement
+{
+private:
+    iDynTree::Vector3 m_VRP; /**< Position of the Virtual Repellent Point */
```

Comment:
I think this is related to simplified models, not to the centroidal linear momentum. I am crying inside :joy: 

I guess this is a possible way for controlling the linear momentum, but it is very dependent on the controller you want to implement at the end, i.e. it is too task-dependent while here you are developing a generic library.

Reply:
You are right. Fixed in 28289339c2bd595457bb2a6ac390828762281e6f